src/codeu/chat/client/commandline/Chat.java

```diff
@@ -27,447 +27,589 @@
+					System.out.format(
+							"CONVERSATION %s (UUID:%s)\n",
+							conversation.conversation.title,
+							conversation.conversation.id);
+				}
+			}
+		});
+
+		// C-ADD (add conversation)
+		//
+		// Add a command that will create and join a new conversation when the user
+		// enters "c-add" while on the user panel.
+		//
+		panel.register("c-add", new Panel.Command() {
+			@Override
+			public void invoke(List<String> args) {
+				final String name = (args.iterator()).hasNext() ? (args.iterator()).next().trim() : "";
+				if (name.length() > 0) {
+					final ConversationContext conversation = user.start(name);
+					if (conversation == null) {
+						System.out.println("ERROR: Failed to create new conversation");
+					} else {
+						panels.push(createConversationPanel(conversation));
+					}
+				} else {
+					System.out.println("ERROR: Missing <title>");
+				}
+			}
+		});
+
+		// C-JOIN (join conversation)
+		//
+		// Add a command that will joing a conversation when the user enters
+		// "c-join" while on the user panel.
+		//
+		panel.register("c-join", new Panel.Command() {
+			@Override
+			public void invoke(List<String> args) {
+				Iterator<String> itr = args.iterator();
+				final String name = itr.hasNext() ? itr.next().trim() : "";
+				if (name.length() > 0) {
+					final ConversationContext conversation = find(name);
+					if (conversation == null) {
+						System.out.format("ERROR: No conversation with name '%s'\n", name);
+					} else {
+						panels.push(createConversationPanel(conversation));
+					}
+				} else {
+					System.out.println("ERROR: Missing <title>");
+				}
+			}
+
+			// Find the first conversation with the given name and return its context.
+			// If no conversation has the given name, this will return null.
+			private ConversationContext find(String title) {
+				for (final ConversationContext conversation : user.conversations()) {
+					if (title.equals(conversation.conversation.title)) {
+						return conversation;
+					}
+				}
+				return null;
+			}
+		});
+
+		// U-INT-ADD (add user interest)
+		//
+		// Add a command that will add another user as an interest
+		// for the user in the user panel.
+		//
+		panel.register("u-int-add", new Panel.Command() {
+			@Override
+			public void invoke(List<String> args) {
+				final String name = (args.iterator()).hasNext() ? (args.iterator()).next().trim() : "";
+				if (name.length() > 0) {
+					final User userInterest = findUser(name);
+					if (userInterest == null) {
+						System.out.format("ERROR: No user with name '%s'\n", name);
+					} else {
+						userInterest.interestedUsers.add(user.user.id);
+					}
+				} else {
+					System.out.println("ERROR: Missing <name>");
+				}
+			}
+
+			//Find user object by name
+			private User findUser(String name) {
+				for (final UserContext userCon : user.users()) {
+					if (name.equals(userCon.user.name)) {
+						return userCon.user;
+					}
+				}
+				return null;
```

Comment:
This works! I believe findUser for the u-int-add and u-int-del are the same? Is it possible to move this method out to the class so that it only needs to be defined once?